cmd/precise-code-intel-api-server/internal/janitor/janitor.go

```diff
@@ -0,0 +1,68 @@
+package janitor
+
+import (
+	"context"
+	"time"
+
+	"github.com/inconshreveable/log15"
+	"github.com/sourcegraph/sourcegraph/cmd/precise-code-intel-api-server/internal/db"
+)
+
+type Janitor struct {
+	db              db.DB
+	janitorInterval time.Duration
+}
+
+type JanitorOpts struct {
+	DB              db.DB
+	JanitorInterval time.Duration
+}
+
+func NewJanitor(opts JanitorOpts) *Janitor {
+	return &Janitor{
+		db:              opts.DB,
+		janitorInterval: opts.JanitorInterval,
+	}
+}
+
+func (j *Janitor) Start() {
```

Comment:
```suggestion
func (j *Janitor) Run() {
```

just a suggestion :-)

Reply:
`janitor.Jan()`

Reply:
I plan on rewriting the janitor to be a generic task runner as we'll have periodic things to do that aren't _just_ cleanup. I'll batch those name changes together :)